### PR TITLE
add connectorx

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by timkpaine** [comment]
@xhochy @isuruf let me know if anything to adjust

File: recipes/connectorx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - cargo-bundle-licenses
+  host:
+    - python
+    - pip
+    - poetry
+    - maturin >=0.12.11,<0.13
+  run:
+    - python
+    - numpy >=1.21.5
+    - libzlib
```
Comment (xhochy) [inline recipes/connectorx/meta.yaml:35]: This indicates that you should have `zlib` in `host`. Then `libzlib` will be automatically added during the build as a `run_exports`

**Comment by timkpaine** [comment]
@xhochy adjusted

File: recipes/connectorx/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -ex
+
+cp $SRC_DIR/README.md $SRC_DIR/connectorx-python/README.md
+cp $SRC_DIR/LICENSE $SRC_DIR/connectorx-python/LICENSE
+
+pushd $SRC_DIR/connectorx-python
+poetry install
```
Comment (xhochy) [inline recipes/connectorx/build.sh:9]: This installs all dependencies of connectorx-python separately from PyPI. Is this really needed? Shouldn't they come from conda-forge?

Comment (timkpaine) [reply recipes/connectorx/build.sh:9]: For pip, env vars are set to force no deps. Idk if this is done for poetry, and I don't know if it's possible

https://github.com/python-poetry/poetry/issues/3377

Comment (timkpaine) [reply recipes/connectorx/build.sh:9]: I can try deleting this line?

File: recipes/connectorx/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+    - posix                               # [win]
+    - cmake
+    - make                                # [unix]
+    - cargo-bundle-licenses
+  host:
+    - python
+    - pip
+    - poetry
```
Comment (Maxyme) [inline recipes/connectorx/meta.yaml:30]: Poetry is a different package manager. Is it possible to use only conda packages and remove poetry from the required list of packages?

**Pull request merged**